store/tikv/region_cache.go

```diff
@@ -673,25 +839,31 @@ func (r *Region) EndKey() []byte {
 	return r.meta.EndKey
 }
 
-// GetContext constructs kvprotopb.Context from region info.
-func (r *Region) GetContext() *kvrpcpb.Context {
-	return &kvrpcpb.Context{
-		RegionId:    r.meta.Id,
-		RegionEpoch: r.meta.RegionEpoch,
-		Peer:        r.peer,
+// switchWorkStore switches current store to the one on specific store. It returns
+// false if no peer matches the storeID.
+func (c *RegionCache) switchWorkStore(r *Region, storeID uint64) (foundLeader bool) {
+	if len(r.meta.Peers) == 0 {
+		return
 	}
-}
 
-// SwitchPeer switches current peer to the one on specific store. It returns
-// false if no peer matches the storeID.
-func (r *Region) SwitchPeer(storeID uint64) bool {
-	for _, p := range r.meta.Peers {
+	leaderIdx := 0
+	for i, p := range r.meta.Peers {
 		if p.GetStoreId() == storeID {
-			r.peer = p
-			return true
+			leaderIdx = i
+			foundLeader = true
 		}
 	}
-	return false
+
```

Comment:
Forgot to check `foundLeader`?

Reply:
this keep old logic, not found leader use idx 0 as current workPeer, `foundLeader` just used to return to caller to check whether leader is update to specific one, it's better rename miss lead name  foundLeader variable.